website/project/views/comment.py

```diff
@@ -15,104 +15,214 @@
 from website.notifications.emails import notify
 from website.filters import gravatar
 from website.models import Guid, Comment
+from website.addons.base import GuidFile
 from website.project.decorators import must_be_contributor_or_public
 from datetime import datetime
 from website.project.model import has_anonymous_link
+from website.project.views.node import _view_project, n_unread_comments
 
 
-def resolve_target(node, guid):
+@must_be_contributor_or_public
+def view_comments_project(**kwargs):
+    """
+    Returns information needed to get comments for the total discussion page
+    """
+    node = kwargs['node'] or kwargs['project']
+    auth = kwargs['auth']
+
+    ret = {
+        'comment_target': 'total',
+        'comment_target_id': None
+    }
+
+    ret.update(_view_project(node, auth, primary=True, check_files=True))
+    return ret
 
+
+@must_be_contributor_or_public
+def view_comments_overview(**kwargs):
+    """
+    Returns information needed to get comments for the discussion overview page
+    """
+    node = kwargs['node'] or kwargs['project']
+    auth = kwargs['auth']
```

Comment:
Put `auth` into the function definition rather than pulling from `kwargs`.

https://cosdev.readthedocs.org/en/latest/osf/guidelines.html#views